src/Microsoft.AspNet.Authentication.Cookies/CookieAppBuilderExtensions.cs

```diff
@@ -20,13 +20,24 @@ public static class CookieAppBuilderExtensions
         /// <param name="configureOptions">Used to configure the options for the middleware</param>
         /// <param name="optionsName">The name of the options class that controls the middleware behavior, null will use the default options</param>
         /// <returns>The original app parameter</returns>
-        public static IApplicationBuilder UseCookieAuthentication([NotNull] this IApplicationBuilder app, Action<CookieAuthenticationOptions> configureOptions = null, string optionsName = "")
+        public static IApplicationBuilder UseCookieAuthentication([NotNull] this IApplicationBuilder app, Action<CookieAuthenticationOptions> configureOptions = null)
         {
             return app.UseMiddleware<CookieAuthenticationMiddleware>(
-                new ConfigureOptions<CookieAuthenticationOptions>(configureOptions ?? (o => { }))
-                {
-                    Name = optionsName
-                });
+                new ConfigureOptions<CookieAuthenticationOptions>(configureOptions ?? (o => { })));
         }
+
+        /// <summary>
+        /// Adds a cookie-based authentication middleware to your web application pipeline.
+        /// </summary>
+        /// <param name="app">The IApplicationBuilder passed to your configuration method</param>
+        /// <param name="configureOptions">Used to configure the options for the middleware</param>
+        /// <param name="optionsName">The name of the options class that controls the middleware behavior, null will use the default options</param>
+        /// <returns>The original app parameter</returns>
+        public static IApplicationBuilder UseCookieAuthentication([NotNull] this IApplicationBuilder app, IOptions<CookieAuthenticationOptions> options)
```

Comment:
One thing to note, this has some bearing on whether we wrap the CookieOptions inside or not
Like what @PinpointTownes suggested:
```
public static IApplicationBuilder UseCookieAuthentication([NotNull] this IApplicationBuilder app, CookieAuthenticationOptions options)
{
    return app.UseMiddleware<CookieAuthenticationMiddleware>(
        new Options<CookieAuthenticationOptions>(options),
        new ConfigureOptions<CookieAuthenticationOptions>(o => { }));
}
```